SeaPublicWebsite.Data/DataAccessProvider.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Microsoft.EntityFrameworkCore;
+
+namespace SeaPublicWebsite.Data;
+
+public class DataAccessProvider : IDataAccessProvider
+{
+    private readonly SeaDbContext context;
+
+    public DataAccessProvider(SeaDbContext context)
+    {
+        this.context = context;
+    }
+    
+    public void AddPropertyData(PropertyData propertyData)
+    {
+        context.PropertyData.Add(propertyData);
+        context.SaveChanges();
+    }
+
+    public void UpdatePropertyData(PropertyData propertyData)
+    {
+        context.PropertyData.Update(propertyData);
+        context.SaveChanges();
+    }
+
+    public async Task<PropertyData> GetSinglePropertyData(string reference)
```

Comment:
Async methods should really have `Async` at the end of their names so that people remember to `await` them